File: Codewars/Python/8 kyu/III & more/Wilson primes.py
def factorial(x):
  resp = 1
  while x>0:
    resp*=x
    x-=1
  return resp

def am_i_wilson(n):
  resp = (factorial(n-1)+1) / (n*n)
  return resp.is_integer()


print(am_i_wilson(5))
print(am_i_wilson(8))
print(am_i_wilson(11))
print(am_i_wilson(13))

File: Codewars/Javascript/8 kyu/II/Wilson primes.js
function amIWilson(p) {
  return (fac(p-1)+1)/(p*p) %1 === 0
}


const fac = (p) => {
  let resp = 1
  while(p>1) resp *=p--
  return resp
}



//console.log(amIWilson(8));//true
console.log(amIWilson(5));//true
console.log(amIWilson(7));//false
console.log(amIWilson(9));//false